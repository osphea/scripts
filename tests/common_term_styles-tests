#!/bin/bash
# Copyright 2018 The Fuchsia Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# Automated tests for //scripts/devshell/lib/common_term_styles.sh
#
# Usage: common_term_styles-tests
#
#   Returns: Error status if actual output does not match expected.

TEST_NAME="$(basename "${BASH_SOURCE[0]}")"
TESTS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"

verbose() {
  echo
  echo "======================================================="
  echo
  echo "$@"
  echo
  "$@"
  echo
}

test_main() {
  local expected_out="${TESTS_DIR}/expected/${TEST_NAME}.out"
  local expected_err="${TESTS_DIR}/expected/${TEST_NAME}.err"
  local capture_dir=$(mktemp -d)
  local actual_out="${capture_dir}/${TEST_NAME}.out"
  local actual_err="${capture_dir}/${TEST_NAME}.err"
  ${TESTS_DIR}/common_term_styles-test-visually 1> "${actual_out}" 2> "${actual_err}"

  local status=0
  verbose diff "${expected_out}" "${actual_out}" || status=$?
  verbose diff "${expected_err}" "${actual_err}" || status=$?

  return $status
}

test_main "$@" || return $?
