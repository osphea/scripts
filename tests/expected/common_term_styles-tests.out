info This is informational
[1;32m[0m
[1;32mINFO: This is informational[0m
[1;32m[0m
warn This is your last warning
[1;33m[0m
[1;33mWARNING: This is your last warning[0m
[1;33m[0m
error Danger! Danger Will Robinson!
[1;91m[0m
[1;91mERROR: Danger! Danger Will Robinson![0m
[1;91m[0m
details
[m  This detail will be[0m
[m  indented from the error.[0m
[m  and could have a link like [4;34mhttps://some/url/here[0m[0m
code
[1;95m    for ( line in lines_of_code ) {[0m
[1;95m      print "this is the demon of code style"[0m
[1;95m    end[0m
